ta4j-core/src/main/java/org/ta4j/core/indicators/ChopIndicator.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * 
+ */
+package org.ta4j.core.indicators;
+
+import org.ta4j.core.num.DoubleNum;
+import org.ta4j.core.num.Num;
+import org.ta4j.core.TimeSeries;
+import org.ta4j.core.indicators.helpers.HighestValueIndicator;
+import org.ta4j.core.indicators.helpers.LowestValueIndicator;
+import org.ta4j.core.indicators.helpers.MaxPriceIndicator;
+import org.ta4j.core.indicators.helpers.MinPriceIndicator;
+
+/**
+ * The "CHOP" index is used to indicate side-ways markets 
+		 see {@link https://www.tradingview.com/wiki/Choppiness_Index_(CHOP)}
+		 100++ * LOG10( SUM(ATR(1), n) / ( MaxHi(n) - MinLo(n) ) ) / LOG10(n)
+				n = User defined period length.
+				LOG10(n) = base-10 LOG of n
+				ATR(1) = Average True Range (Period of 1)
+				SUM(ATR(1), n) = Sum of the Average True Range over past n bars 
+				MaxHi(n) = The highest high over past n bars
+				
+				++ usually this index is between 0 and 100, but could be scalled differently by the 'scaleTo' arg of the constructor
+ *
+ */
+public class ChopIndicator extends CachedIndicator<Num> {
+
+	private ATRIndicator atrIndicator;
+	TimeSeries timeseries;
+	private int timeFrame;
+	public final double LOG10n;
+	double MaxHi = 0-Float.MAX_VALUE, MinLo = Float.MAX_VALUE;
+	private HighestValueIndicator hvi;
+	private LowestValueIndicator lvi;
+	private final double scaleUpTo;
+
+	/**
+	 * ctor
+	 * @param timeseries

```

Comment:
Update Javadoc please, tags should not be empty. Just add something like
` @param timeseries the time series` or ` @param timeseries the {@link TimeSeries}`